src/desdemona/query.clj

```diff
@@ -35,16 +58,28 @@
        (finally
          (in-ns (ns-name old-ns)))))))
 
+(def dsl-literal?
+  "Is this a literal in the DSL?"
+  string?)
+
 (defn ^:private dsl->logic
   "Given a DSL query, compile it to the underlying logic (miniKanren)
   expressions."
   [dsl-query]
   (m/match [dsl-query]
-    [((= ((attr lvar) :seq) value) :seq)]
-    `(l/featurec ~lvar {~attr ~value})
-
-    [((= value ((attr lvar) :seq)) :seq)]
-    `(l/featurec ~lvar {~attr ~value})
+    [(('= & terms) :seq)]
+    (let [{literals true attr-terms false} (group-by dsl-literal? terms)
+          featurec-term (fn [value [attr lvar]]
+                          (let [lvar (free-sym lvar)]
+                            `(l/featurec ~lvar {~attr ~value})))
+          literal-featurec-term (partial featurec-term (first literals))]
+      (m/match [(count literals) (count attr-terms)]
+        [1 1] (literal-featurec-term (first attr-terms))
+        [1 _] (let [logic-terms (map literal-featurec-term attr-terms)]
+                `(l/all ~@logic-terms))
+        [0 _] (let [u (gensym)
+                    logic-terms (map (partial featurec-term u) attr-terms)]
+                `(l/fresh [~u] ~@logic-terms))))
```

Comment:
FYI: you could write this as one clause, but then it would change what all of the old `(= (:ip x) "10.0.0.1")` generated code would look like.